src/serializer/ResidualHeapSerializer.js

```diff
@@ -1161,6 +1221,83 @@ export class ResidualHeapSerializer {
+          // result -- ignore TODO: return the result from the function somehow
+          // Generator -- visit all entries
+          // Bindings -- only need to serialize bindings if they're captured by some nested function ??
+          //          -- need to apply them and maybe need to revisit functions in ancestors to make sure
+          //          -- we don't overwrite anything they capture
+          //          -- TODO: deal with these properly
+          // PropertyBindings -- visit any property bindings that aren't to createdobjects
+          // CreatedObjects -- should take care of itself
+          this.realm.applyEffects([r, new Generator(this.realm), ob, pb, co]);
+          // Allows us to emit function declarations etc. inside of this additional
+          // function instead of adding them at global scope
+          // TODO: make sure this generator isn't getting mutated oddly
+          this.additionalFunctionValueNestedFunctions = ((nestedFunctions: any): Set<FunctionValue>);
+          let emitter = this.emitter;
+          let serializePropertiesAndBindings = () => {
+            for (let propertyBinding of pb.keys()) {
+              let binding: PropertyBinding = ((propertyBinding: any): PropertyBinding);
+              let object = binding.object;
+              // These should be in the generator so we can skip them
+              if (object.isPartialObject()) continue;
+              invariant(object instanceof ObjectValue);
+              if (co.has(object)) continue; // Created Object's binding
+              if (object.intrinsicName === "global") continue; // Avoid double-serialization
+              if (object.refuseSerialization) continue; // modification to internal state
+              this._emitProperty(object, binding.key, binding.descriptor, true);
+            }
+            for (let binding of ob.keys()) {
+              let old_value = ob.get(binding);
+              let new_value = binding.value;
+              invariant(old_value !== new_value);
+              if (old_value) {
+                let lhs = ((this.serializeValue(old_value): any): BabelNodeIdentifier);
+                invariant(t.isIdentifier(lhs));
+                emitter.emit(
+                  t.expressionStatement(
+                    t.assignmentExpression(
+                      "=",
+                      lhs,
+                      new_value ? this.serializeValue(new_value) : t.valueToNode(undefined)
+                    )
+                  )
+                );
+              }
+            }
+          };
+          let body = this._serializeAdditionalFunction(g, serializePropertiesAndBindings);
+          invariant(body.length > 0, "An additional function has no body");
+          invariant(additionalFunctionValue instanceof ECMAScriptSourceFunctionValue);
+          rewrittenAdditionalFunctions.set(additionalFunctionValue, body);
+          // re-resolve initialized modules to include things from additional functions
```

Comment:
Why include things from additional functions? Nobody can rely on this.

Reply:
This is mainly just so that I can count how many modules were initialized by additional functions. If you want I can remove this.